circuit/src/circuit.rs

```diff
@@ -85,8 +85,23 @@ impl Default for WormholeCircuit {
 }
 
 impl WormholeCircuit {
-    pub fn new() -> Self {
-        Self::default()
+    pub fn new(zk: bool) -> Self {
```

Comment:
I'm not a big fan of configuring zk via this flag. Instead, how about passing in the full configuration, delegating the job of configuring the circuit to the caller? We could then have some default configuration that's dependent on if we're in a testing environment or not.

Reply:
I don't love it either. Your idea seems reasonable.